bundles/org.eclipse.swt/Eclipse SWT/win32/org/eclipse/swt/widgets/Tree.java

```diff
@@ -4347,7 +4348,11 @@ void setItemCount (int count, long hParent) {
 /*public*/ void setItemHeight (int itemHeight) {
 	checkWidget ();
 	if (itemHeight < -1) error (SWT.ERROR_INVALID_ARGUMENT);
-	OS.SendMessage (handle, OS.TVM_SETITEMHEIGHT, itemHeight, 0);
+	long newHeight = OS.SendMessage (handle, OS.TVM_SETITEMHEIGHT, itemHeight, 0);
+	// if the item height was set at least once programmatically, it will never be
+	// done by the OS anymore. To handle resizing after DPI changes, this must be
+	// tracked with the itemHeightSetProgrammatically-flag
```

Comment:
It's kind of confusing what this comment wants to say, can you rephrase it?

If I interpret the changes correctly, the `if` in line 8288 can only work if the method `setItemHeight(int)` has been called at least twice, right?

Reply:
Sure, I will rephrase it a bit

But why twice?